sklearn/preprocessing/_label.py

```diff
@@ -111,7 +115,10 @@ def _encode(values, uniques=None, encode=False, check_unknown=True):
         try:
             res = _encode_python(values, uniques, encode)
         except TypeError:
-            raise TypeError("argument must be a string or number")
+            raise TypeError("Encoders require their input to be strings or "
+                            "numbers. Got {}")
```

Comment:
{} should be substituted with an appropriate value... Or we should leave it out.

Reply:
Oh my... 

I think we should leave it out because otherwise we would have to find all the input types in the input